deltacat/utils/common.py

```diff
@@ -1,25 +1,46 @@
+import argparse
 import hashlib
 import os
 import time
+import sys
 from typing import Any, Dict
 
 
-def env_bool(key: str, default: bool) -> int:
+def get_cli_arg(key: str) -> Any:
+    parser = argparse.ArgumentParser()
+
+    # allow hyphens in keys, normalize as underscores
+    normalized_key = key.replace("-", "_")
+
+    parser.add_argument(f"--{key}", metavar=normalized_key, type=str)
+    args, _ = parser.parse_known_args(sys.argv[1:])  # Allow unknown args
+    return getattr(args, normalized_key, None)
+
+
+def env_bool(key: str, default: bool) -> bool:
+    cli_value = get_cli_arg(key)
```

Comment:
Hmm... since you can also set environment variables within the context of a single bash command, can we just let users do that and not overload the environment variables here to either look for CLI args or environment variables?

(i.e., `$ METAFILE_FORMAT=json python ./my-script.py`)

We can then better create a more formal/unified deltacat CLI experience within the context of a dedicated CLI application, similar to `dev/deploy/aws/scripts/runner.py`.